### PR TITLE
Upgrade dependencies

### PR DESCRIPTION
**Pull request by raymondfeng**
### Description

Update to loopback-swagger@4.x

#### Related issues

- None

### Checklist

- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the [style
  guide](http://loopback.io/doc/en/contrib/style-guide.html)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "loopback-api-definition",
-  "version": "3.0.0",
+  "version": "4.0.0-alpha.1",
```
Comment (bajtos) [inline package.json]: I think there is no need to bump up the major version, considering that 
 - `loopback-api-definition` depends only on `specgen` part of `loopback-swagger`
 - the breaking changes in `loopback-swagger@4` were related only to `codegen`

File: package.json [hunk 2]
```diff
@@ -25,15 +25,15 @@
...
-    "eslint": "^2.8.0",
-    "eslint-config-loopback": "^2.0.0",
-    "loopback": "^1.7.4",
-    "loopback-datasource-juggler": "^1.3.10",
-    "mocha": "^1.18.2"
+    "chai": "^3.5.0",
+    "eslint": "^3.16.1",
+    "eslint-config-loopback": "^8.0.0",
+    "loopback": "^3.4.0",
+    "loopback-datasource-juggler": "^3.2.0",
```
Comment (bajtos) [inline package.json]: This is no longer needed in `loopback@3`, juggler comes as a regular loopback dependency now.

File: package.json (hunk 1 above)
Comment (raymondfeng) [inline package.json]: Making sense.

**Comment by raymondfeng** [comment]
Comments addressed.

**Pull request merged**